packages/hash/api/src/model/knowledge/entity.model.ts

```diff
@@ -112,6 +118,139 @@ export default class {
     });
   }
 
+  /**
+   * Create an entity along with any new/existing entities specified through links.
+   *
+   * @param params.createdById the account id that is creating the entity
+   * @param params.entityDefinition the definition of how to get or create the entity optionally with linked entities
+   */
+  static async createEntityWithLinks(
+    graphApi: GraphApi,
+    params: {
+      createdById: string;
+      entityDefinition: KnowledgeEntityDefinition;
+    },
+  ): Promise<EntityModel> {
+    const { createdById, entityDefinition: entityDefinitions } = params;
+
+    const entitiesInTree = linkedTreeFlatten<
+      KnowledgeEntityDefinition,
+      KnowledgeLinkedEntityDefinition,
+      "linkedEntities",
+      "entity"
+    >(entityDefinitions, "linkedEntities", "entity");
+
+    const entities = await Promise.all(
+      entitiesInTree.map(async (entityDefinition) => ({
```

Comment:
Nit: I would prefer if we don't rename the parameter like this, maybe let's try:

```suggestion
    const { createdById, entityDefinition } = params;

    const entitiesInTree = linkedTreeFlatten<
      KnowledgeEntityDefinition,
      KnowledgeLinkedEntityDefinition,
      "linkedEntities",
      "entity"
    >(entityDefinition, "linkedEntities", "entity");

    const entities = await Promise.all(
      entitiesInTree.map(async (definition) => ({
```